lib/Frontend/CompilerInvocation.cpp

```diff
@@ -401,9 +402,17 @@ static void SaveModuleInterfaceArgs(ModuleInterfaceOptions &Opts,
     return;
   ArgStringList RenderedArgs;
   ArgStringList RenderedArgsIgnorable;
+  ArgStringList RenderedArgsIgnorablePrivate;
   for (auto A : Args) {
     if (A->getOption().hasFlag(options::ModuleInterfaceOptionIgnorable)) {
-      A->render(Args, RenderedArgsIgnorable);
+      StringRef str = A->getValue();
+      if (A->getOption().getName().startswith("package-name")) {
```

Comment:
I think it would be worthwhile to instead have something like `options::ModuleInterfaceOptionIgnorablePrivateOnly` and have that apply to `-package-name` instead of string matching here.